R/course.R

```diff
@@ -236,21 +240,22 @@ use_zip <- function(url,
 NULL
 
 # 1. downloads from `url`
-# 2. determines filename from content-description header (with fallbacks)
-# 3. returned path has content-type and content-description as attributes
+# 2. calls a retry-capable helper to download the ZIP file
+# 3. determines filename from content-description header (with fallbacks)
+# 4. returned path has content-type and content-description as attributes
 tidy_download <- function(url, destdir = getwd()) {
   check_path_is_directory(destdir)
   tmp <- file_temp("tidy-download-")
-  h <- curl::new_handle(noprogress = FALSE, progressfunction = progress_fun)
-  curl::curl_download(url, tmp, quiet = FALSE, mode = "wb", handle = h)
+
+  h <- download_url(url, destfile = tmp)
```

Comment:
We need the handle `h` here in order to query its content-disposition header (used for naming the destination directory) and content type.